sycl/include/CL/sycl/handler.hpp

```diff
@@ -386,7 +386,19 @@ class __SYCL_EXPORT handler {
                        static_cast<int>(AccessTarget), ArgIndex);
   }
 
-  template <typename T> void setArgHelper(int ArgIndex, T &&Arg) {
+  template <typename T> struct ShouldEnableSetArgHelper {
+    static constexpr bool value = std::is_trivially_copyable<T>::value
```

Comment:
@bader  Should we also check for `std::is_trivially_destructible`?

Reply:
I don't think so, but I'm interested to see the case when it can matter.

Reply:
You are right. trivially copyable implies trivially destructible